extension/intents/slideshow/contentScript.js

```diff
@@ -0,0 +1,347 @@
+          thumbnail.addEventListener("click", thumbnailClick);
+
+          break;
+        }
+        case "IMG": {
+          thumbnailElement.className =
+            "fv-image-thumbnail-container fv-thumbnail";
+
+          thumbnail = iframeDoc.createElement(element.tagName);
+          thumbnail.className = "fv-image-thumbnail";
+          thumbnail.setAttribute("src", element.getAttribute("src"));
+          thumbnail.setAttribute("data-thumb-index", index);
+          thumbnail.addEventListener("click", thumbnailClick);
+
+          break;
+        }
+        case "IFRAME": {
+          thumbnailElement.className =
+            "fv-iframe-thumbnail-container fv-thumbnail";
+
+          iframeBlocker = iframeDoc.createElement("div");
+          iframeBlocker.style.position = "absolute";
+          iframeBlocker.style.top = 0;
+          iframeBlocker.style.left = 0;
+          iframeBlocker.style.zIndex = 2;
+          iframeBlocker.style.width = "100%";
+          iframeBlocker.style.height = "100%";
+          iframeBlocker.setAttribute("data-thumb-index", index);
+          iframeBlocker.addEventListener("click", thumbnailClick);
+          iframeBlocker.style.cursor = "pointer";
+          thumbnailElement.append(iframeBlocker);
+
+          thumbnail = iframeDoc.createElement(element.tagName);
+          thumbnail.className = "fv-iframe-thumbnail";
+          thumbnail.setAttribute("src", element.getAttribute("src"));
+
+          break;
+        }
+      }
+
+      thumbnailElement.append(thumbnail);
+      galleryElements.push(thumbnailElement);
+    });
+  }
+
+  // event handler to switch from selected thumbnail to slide
+  function thumbnailClick(event) {
+    event.preventDefault();
+    const thumbIndex = parseInt(event.target.dataset.thumbIndex, 10);
+    showSlide(thumbIndex);
+    toggleGallery();
+  }
+
+  // toggle between gallery and slide
+  function toggleGallery(event) {
+    if (!event || event.target.className === "fv-view-slide") {
+      galleryContainer.style.display = "none";
+      slideshowContainer.style.display = "flex";
+    } else if (event.target.className === "fv-view-gallery") {
+      galleryContainer.style.display = "block";
+      slideshowContainer.style.display = "none";
+    }
+  }
+
+  function buildSlideStructure() {
+    // create iframe element
+    iframeContainer = document.createElement("iframe");
+    iframeContainer.className = "fv-slideshow-frame";
+    iframeContainer.height = document.documentElement.clientHeight;
+    iframeContainer.width = document.documentElement.clientWidth;
+    iframeContainer.style.display = "none";
+    iframeContainer.style.position = "fixed";
+    iframeContainer.style.top = 0;
+    iframeContainer.style.left = 0;
+    iframeContainer.style.zIndex = 100;
+
+    document.body.appendChild(iframeContainer);
+
+    iframeContainer.addEventListener("load", function() {
+      // contentWindow is first accessible here
```

Comment:
And, a very obscure security check that pertains only to iframes created by webextensions – here you want to verify that the iframe still has the same src, like `if (iframeContainer.documentElement.documentURI !== browser.runtime.getURL("...")) {throw new Error(...)}`

Reply:
Added the security check